### PR TITLE
TCTS telemetry | Last request reference

### PR DESCRIPTION
**Pull request by sudiksha27**
**Goals (and why)**:
Keep information of last request that set the lowerBound of timestamps

### CODE REVIEW

**Review by jeremyk-91 — COMMENTED** [review]
Some of the tests are still failing. Otherwise this looks pretty much like what I'd expect.

File: lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java [hunk 1]
```diff
@@ -127,26 +134,39 @@ public ConjureStartTransactionsResponse startTransactions(ConjureStartTransactio
...
+                .boundFromTransactions(lowerBoundFromTransaction.get())
+                .build();
     }
 
     private void checkTimestamp(
             TimestampBounds bounds, OperationType type, long lowerFreshTimestamp, long upperFreshTimestamp) {
-        if (lowerFreshTimestamp <= Math.max(bounds.boundFromTimestamps(), bounds.boundFromTransactions())) {
+        if (lowerFreshTimestamp <= bounds.getMaxLowerBound()) {
             timestampViolationCallback.run();
             throw clocksWentBackwards(bounds, type, lowerFreshTimestamp, upperFreshTimestamp);
```
Comment (gmaretic) [inline lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java:171]: We don't seem to be doing anything with the new information. Surely we should at least log the three bounds with timestamps, the timestamp of the comparison, maybe the current timestamp etc?

File: lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java [hunk 2]
```diff
@@ -127,26 +134,39 @@ public ConjureStartTransactionsResponse startTransactions(ConjureStartTransactio
             ToLongFunction<T> lowerBoundExtractor,
             ToLongFunction<T> upperBoundExtractor,
             OperationType operationType) {
+        // take snapshot before making the request
+        Instant wallClockTimeBeforeRequest = Instant.now();
```
Comment (gmaretic) [inline lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java:138]: Probably want to take the time snapshot after we get the bounds? Awkward if the bounds update in between

**Review by gmaretic — COMMENTED** [review]
The change makes sense, but we don't seem to be using the new stuff anywhere. Also, maybe we want to inject our own `Clock` so we can do some testing based on that and verify also that the logged instants make sense?
Maybe have the callback be a consumer of records and current time or some such?

File: lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java (hunk 1 above)
Comment (sudiksha27) [reply lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java:171]: Most of these was being done but added more logging for wall clock times of current request

File: lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java (hunk 2 above)
Comment (sudiksha27) [reply lock-api/src/main/java/com/palantir/lock/client/TimestampCorroboratingTimelockService.java:138]: This is also happening but I can make it immediately after we get the response.

**Comment by gmaretic** [comment]
👍

**Pull request merged**